From 86fe551778c3e3e8fcf54b8800196c11309f158d Mon Sep 17 00:00:00 2001
From: AngeloGioacchino Del Regno <angelogioacchino.delregno@somainline.org>
Date: Wed, 20 Jan 2021 15:27:42 +0100
Subject: [PATCH 021/127] arm64: dts: msm8998: Wire up interconnects and OPPs
 to sdhci port 2

Wire up the OPP table and interconnects to the SDHCI port 2 to
improve performance and power consumption.

Signed-off-by: AngeloGioacchino Del Regno <angelogioacchino.delregno@somainline.org>
---
 arch/arm64/boot/dts/qcom/msm8998.dtsi | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/msm8998.dtsi b/arch/arm64/boot/dts/qcom/msm8998.dtsi
index 26a77d5959fe..3389957032dd 100644
--- a/arch/arm64/boot/dts/qcom/msm8998.dtsi
+++ b/arch/arm64/boot/dts/qcom/msm8998.dtsi
@@ -2136,6 +2136,7 @@ sdhc2: mmc@c0a4900 {
 			reg = <0x0c0a4900 0x314>, <0x0c0a4000 0x800>;
 			reg-names = "hc", "core";
 
+			interconnects = <&a2noc MASTER_SDCC_2 &a2noc SLAVE_A2NOC_SNOC>;
 			interrupts = <GIC_SPI 125 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 221 IRQ_TYPE_LEVEL_HIGH>;
 			interrupt-names = "hc_irq", "pwr_irq";
@@ -2146,6 +2147,26 @@ sdhc2: mmc@c0a4900 {
 				 <&xo>;
 			bus-width = <4>;
 			status = "disabled";
+
+			sdhc2_opp_table: sdhc2-opp-table {
+				 compatible = "operating-points-v2";
+
+				 opp-50000000 {
+					opp-hz = /bits/ 64 <50000000>;
+					opp-peak-kBps = <400000>;
+					opp-avg-kBps = <200000>;
+				 };
+				 opp-100000000 {
+					opp-hz = /bits/ 64 <100000000>;
+					opp-peak-kBps = <800000>;
+					opp-avg-kBps = <400000>;
+				 };
+				 opp-200000000 {
+					opp-hz = /bits/ 64 <200000000>;
+					opp-peak-kBps = <800000>;
+					opp-avg-kBps = <800000>;
+				 };
+			};
 		};
 
 		blsp1_dma: dma-controller@c144000 {
-- 
2.38.1

